Explorer/Assets/DCL/PluginSystem/World/TextShapePlugin.cs

```diff
@@ -5,55 +5,49 @@
+using TMPro;
+using UnityEngine;
 
 namespace DCL.PluginSystem.World
 {
     public class TextShapePlugin : IDCLWorldPlugin
     {
-        private readonly ITextShapeRendererFactory textShapeRendererFactory;
         private readonly IPerformanceBudget instantiationFrameTimeBudgetProvider;
         private readonly IComponentPoolsRegistry componentPoolsRegistry;
+        private readonly IFontsStorage fontsStorage;
 
-        public TextShapePlugin(IPerformanceBudget instantiationFrameTimeBudgetProvider, IComponentPoolsRegistry componentPoolsRegistry, IPluginSettingsContainer settingsContainer) : this(
-            instantiationFrameTimeBudgetProvider,
-            componentPoolsRegistry,
-            settingsContainer.GetSettings<FontsSettings>().AsCached()
-        ) { }
+        private readonly MaterialPropertyBlock materialPropertyBlock = new ();
 
-        public TextShapePlugin(IPerformanceBudget instantiationFrameTimeBudgetProvider, IComponentPoolsRegistry componentPoolsRegistry, IFontsStorage fontsStorage) : this(
-            new PoolTextShapeRendererFactory(componentPoolsRegistry, fontsStorage),
-            instantiationFrameTimeBudgetProvider,
-            componentPoolsRegistry
-        ) { }
-
-        public TextShapePlugin(ITextShapeRendererFactory textShapeRendererFactory, IPerformanceBudget instantiationFrameTimeBudgetProvider, IComponentPoolsRegistry componentPoolsRegistry)
+        public TextShapePlugin(IPerformanceBudget instantiationFrameTimeBudgetProvider, IComponentPoolsRegistry componentPoolsRegistry, IPluginSettingsContainer settingsContainer)
         {
-            this.textShapeRendererFactory = textShapeRendererFactory;
+            fontsStorage = settingsContainer.GetSettings<FontsSettings>().AsCached();
             this.instantiationFrameTimeBudgetProvider = instantiationFrameTimeBudgetProvider;
             this.componentPoolsRegistry = componentPoolsRegistry;
+
+            componentPoolsRegistry.AddGameObjectPool<TextMeshPro>();
         }
 
+        public UniTask Initialize(IPluginSettingsContainer container, CancellationToken ct) =>
+            UniTask.CompletedTask;
+
         public void Dispose()
         {
             //ignore
         }
 
-        public UniTask Initialize(IPluginSettingsContainer container, CancellationToken ct) =>
-            UniTask.CompletedTask;
-
         public void InjectToWorld(ref ArchSystemsWorldBuilder<Arch.Core.World> builder, in ECSWorldInstanceSharedDependencies sharedDependencies, in PersistentEntities persistentEntities, List<IFinalizeWorldSystem> finalizeWorldSystems, List<ISceneIsCurrentListener> sceneIsCurrentListeners)
         {
-            InstantiateTextShapeSystem.InjectToWorld(ref builder, textShapeRendererFactory, instantiationFrameTimeBudgetProvider);
-            UpdateTextShapeSystem.InjectToWorld(ref builder);
+            IComponentPool<TextMeshPro> textMeshProPool = componentPoolsRegistry.GetReferenceTypePool<TextMeshPro>();
```

Comment:
Don't request the pool second time: `componentPoolsRegistry.AddGameObjectPool<TextMeshPro>();` already returns a reference to the pool: just introduce a `private readonly` field